modules/services/mapillary.js

```diff
@@ -236,11 +236,32 @@ function searchLimited(psize, limit, projection, rtree) {
     limit = limit || 3;
 
     var partitions = partitionViewport(psize, projection);
-    return _.flatten(_.compact(_.map(partitions, function(extent) {
+    console.time('previous');
+
+    var x =  _.flatten(_.map(partitions, function(extent) {
         return rtree.search(extent.bbox())
             .slice(0, limit)
             .map(function(d) { return d.data; });
-    })));
+    }));
+    console.timeEnd('previous');
+
+    console.time('new');
+    var a = partitions.reduce(function(prev, curr) {
```

Comment:
@bhousel I have tried to squeeze all the operations in one `reduce` hoping for a better performance over code legibility, thoughts?

Reply:
oh this looks great!  `reduce` is perfectly readable  👍